src/test/rustdoc/const-display.rs

```diff
@@ -67,3 +67,20 @@ impl Foo {
     #[rustc_const_stable(feature = "rust1", since = "1.2.0")]
     pub const fn stable_impl() -> u32 { 42 }
 }
+
+#[stable(feature = "rust1", since = "1.0.0")]
+pub struct Bar;
+
+impl Bar {
+    // Do not show non-const stabilities that are the same as the enclosing item.
+    // @matches 'foo/struct.Bar.html' '//span[@class="since"]' '^const: 1.2.0$'
+    #[stable(feature = "rust1", since = "1.0.0")]
+    #[rustc_const_stable(feature = "rust1", since = "1.2.0")]
+    pub const fn stable_impl() -> u32 { 42 }
+
+    // Show const-stability even for unstable functions.
+    // @matches 'foo/struct.Bar.html' '//span[@class="since"]' '^const: 1.3.0$'
+    #[unstable(feature = "foo2", issue = "none")]
+    #[rustc_const_stable(feature = "rust1", since = "1.3.0")]
```

Comment:
I think this is invalid, you cannot have a `rustc_const_stable` with an `unstable` attribute, but due to a bug in the compiler it's currently accepted despite being obviously wrong.

Reply:
I disagree that it's "obviously wrong", allowing unstable functions to be called in a const-stable context is [called out as a use-case in the rustc-guide](https://rustc-dev-guide.rust-lang.org/stability.html#rustc_const_stable).

Reply:
I agree that `unstable` + `rustc_const_stable` doesn't make a lot of sense; `stable` functions can call `unstable` functions, so shouldn't `rustc_const_stable` functions be able to call `rustc_const_unstable` functions?

Reply:
Regardless of whether const-stable unstable functions are disallowed in the future, the implementation of this PR won't be affected: the invalid test case will just be removed.

For now, it's supported, so it makes sense to me to show it in the documentation.